### PR TITLE
Refactoring of evaluator managers

### PR DESCRIPTION
**Pull request by anglee**
This batch of changes includes:
- [evaluatorManager](https://github.com/twosigma/beaker-notebook/blob/release-refact/core/src/main/web/js/module/evaluatorManager.js) has be separated and completely rewritten into [bkEvaluatorManager](https://github.com/twosigma/beaker-notebook/blob/dev-refact-evaluator-managers/core/src/main/web/js/module/bkEvaluatorManager.js) and [bkEvaluatePluginManager](https://github.com/twosigma/beaker-notebook/blob/dev-refact-evaluator-managers/core/src/main/web/js/module/bkEvaluatePluginManager.js)
- the evaluator plugins format is changed to use AMD module style (requireJS)
- when adding a plugin by clicking button in the [bkNotebookEvaluators](https://github.com/twosigma/beaker-notebook/blob/dev-refact-evaluator-managers/core/src/main/web/js/module/bkNotebookEvaluators.js) (plugin manager), the notebook model store the plugin as plugin name instead of plugin urls, also, all the default notebooks and tutorial notebooks are changed to refer plugin with name instead of url

### CODE REVIEW

File: plugin/ipythonPlugins/src/dist/ipython/ipython.js [hunk 1]
```diff
@@ -271,4 +269,20 @@
         onSuccess, onFail);
   };
   init();
-})();
+
+  exports.getEvaluatorFactory = function(settings) {
```
Comment (scottdraves) [inline plugin/ipythonPlugins/src/dist/ipython/ipython.js]: this looks like a pretty complicated piece of boilerplate.
can it be factored into bkHelper?

File: core/src/main/web/js/module/bkEvaluatePluginManager.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+              loadingInProgressPlugins = _(loadingInProgressPlugins).filter(function(it) {
+                return it.url !== url;
+              });
+            });
+
+            return deferred.promise;
+          }
+        },
+        createEvaluatorThenExit: function(settings) {
+          console.log("createEvaluatorThenExit", settings.plugin);
```
Comment (scottdraves) [inline core/src/main/web/js/module/bkEvaluatePluginManager.js]: pls remove debug console logging

File: plugin/ipythonPlugins/src/dist/ipython/ipython.js (hunk 1 above)
Comment (anglee) [inline plugin/ipythonPlugins/src/dist/ipython/ipython.js]: I have simplified the code.

Comment (scottdraves) [inline plugin/ipythonPlugins/src/dist/ipython/ipython.js]: OK that's better but it's still the same 9 lines of code in every plugin.
Why not just a 1-line call to a function?

**Pull request merged**